src/GraphQL/Validation/DocumentValidator.cs

```diff
@@ -71,6 +71,14 @@ public class DocumentValidator : IDocumentValidator
                 schema.Initialize();
             }
 
+            if (rules == null)
+            {
+                rules = CoreRules;
+            }
+
+            if (!rules.Any())
+                return SuccessfullyValidatedResult.Instance;
```

Comment:
This will increase performance when an empty ruleset is passed to a document executer.  For example, used when a cached document without variables has already been validated.  My own projects will benefit from this change immediately.

Reply:
Of course.